beams/bin/run_main.py

```diff
@@ -3,24 +3,91 @@
 """
 
 import logging
+import time
+from functools import partial
 from pathlib import Path
 
+from py_trees.console import read_single_keypress
+from py_trees.display import unicode_blackboard, unicode_tree
+from py_trees.trees import BehaviourTree
+from py_trees.visitors import SnapshotVisitor
+
+from beams.logging import LoggingVisitor
 from beams.tree_config import get_tree_from_path
 
 logger = logging.getLogger(__name__)
 
 
-def main(filepath: str):
+def snapshot_post_tick_handler(
+    snapshot_visitor: SnapshotVisitor,
+    show_tree: bool,
+    show_blackboard: bool,
+    behaviour_tree: BehaviourTree,
+) -> None:
+    """
+    Print data about the part of the tree visited.
+    Does not log, to allow use of color codes.
+    ``snapshot_visitor`` keeps track of visited nodes and their status
+
+    Args:
+        snapshot_handler: gather data about the part of the tree visited
+        behaviour_tree: tree to gather data from
+    """
+    if show_tree:
+        print(
+            "\n"
+            + unicode_tree(
+                root=behaviour_tree.root,
+                visited=snapshot_visitor.visited,
+                previously_visited=snapshot_visitor.previously_visited,
+                show_status=True
+            )
+        )
+
+    if show_blackboard:
+        print(unicode_blackboard())
+
+
+def main(
+    filepath: str,
+    tick_count: int,
+    tick_delay: float,
+    interactive: bool,
+    show_node_status: bool,
+    show_tree: bool,
+    show_blackboard: bool,
+):
     logger.info(f"Running behavior tree at {filepath}")
     # grab config
     fp = Path(filepath).resolve()
     if not fp.is_file():
         raise ValueError("Provided filepath is not a file")
 
     tree = get_tree_from_path(fp)
-    print(tree)
-    # TODO: the rest of whatever we determine the "run" process to be
-    # run external server?
-    # setup tree?
-    # tick?
-    # settings for ticking? (continuous tick?, as separate process?)
+    tree.visitors.append(LoggingVisitor(print_status=show_node_status))
+
+    snapshot_visitor = SnapshotVisitor()
+    tree.add_post_tick_handler(
+        partial(snapshot_post_tick_handler,
+                snapshot_visitor,
+                show_tree,
+                show_blackboard)
+    )
+    tree.setup()
+    if tick_count <= 0:
+        while True:
+            try:
+                tree.tick()
+                time.sleep(tick_delay)
```

Comment:
The infinite version of this ignores the user's `interactive` cli arg:
```suggestion
                if interactive:
                    read_single_keypress()
                else:
                    time.sleep(tick_delay)
```

Reply:
Sigh, never code while distracted